docs/en/operations/external-authenticators/ssl-x509.md

```diff
@@ -0,0 +1,24 @@
+# SSL X.509 certificate authentication {#ssl-external-authentication}
+
+[SSL 'strict' option](../server-configuration-parameters/settings.md#server_configuration_parameters-openssl) enables mandatory certificate validation for the incoming connections. In this case, only connections with trusted certificates can be established. Connections with untrusted certificates will be rejected. Thus, certificate validation allows to uniquely authenticate an incoming connection. `Common Name` field of the certificate is used to identify connected user. This allows to associate multiple certificates with the same user. Additionally, reissuing and revoking of the certificates does not affect the ClickHouse configuration.
+
+To enable SSL certificate authentication, a list of `Common Name`'s for each ClickHouse user must be sspecified in the settings file `config.xml `:
+
+**Example**
+```xml
+<clickhouse>
+    <!- ... -->
+    <users>
+        <user_name>
+            <certificates>
+                <common_name>host.domain.com:example_user</common_name>
+                <common_name>host.domain.com:example_user_dev</common_name>
+                <!-- More names -->
+            </certificates>
+            <!-- Other settings -->
+        </user_name>
+    </users>
+</clickhouse>
```

Comment:
It's looks like a very manual thing for now. In more advanced phases it should be able to cooperate with LDAP, for ex. like that:

https://docs.confluent.io/platform/current/kafka/configure-mds/mutual-tls-auth-rbac.html#deep-dive-ssl-principal-mapping-rules

(i.e. extract CN from cert, and do lookup of that in LDAP to understand if it's allowed and what access should the guy get).

Reply:
Not sure it should work that way.
LDAP is not supposed to be aware of the CNs. Instead, ClickHouse should resolve username _from_ the CN, and optionally use LDAP with that username for further steps.
The main goal of the certificate authentication is to avoid passwords and delegate this task to the SSL library.

Reply:
> ClickHouse should resolve username from the CN

That means you have 2 sources of truth for users (i.e. you need to create a user both in LDAP and in ClickHouse config). 
It doesn't fit the requirements of huge companies (they want to maintain the user list in one place), and ClickHouse just follows that.

>  The main goal of the certificate authentication is to avoid passwords and delegate this task to the SSL library.

The mode you created is totally ok and will work great for simpler setups. The use-case I'm referring to will also be needed. I can be added later, we just need to be ready for that.